components/ciphertool/src/main/java/org/wso2/ciphertool/utils/Utils.java

```diff
@@ -151,6 +153,42 @@ public static void writeToPropertyFile(Properties properties, String filePath) {
         }
     }
 
+    /**
+     * Retrieve the value for the given configuration from the following order of files.
+     * 1. deployment.toml
+     * 2. default.json
+     * 3. carbon.xml
+     *
+     * @param config        configuration value
+     * @param key           key of the configuration value in default.json
+     * @param defaultMap    map that contains values from default.json
+     * @param element       element
+     * @param xPath         xpath
+     * @return  configuration value
+     */
+    public static String getValueFromConfigs(String config, String key, Map<String, Object> defaultMap,
+                                             Element element, String xPath) {
+        String value = config;
+        try {
+            // if the value is empty in deployment.toml
+            if (StringUtils.isEmpty(value)) {
+                // read from default.json
+                value = defaultMap.get(key).toString();
+            }
+            // if the value is given as a reference
+            if (value.startsWith("$ref")) {
+                // read from default.json
+                String reference = value.substring(5, value.indexOf('}'));
+                return defaultMap.get(reference).toString();
+            }
+            return value;
+        } catch (NullPointerException e) {
+            // read from carbon.xml if default.json is not available
+            System.err.println("Invalid value " + key + " " + e);

```

Comment:
Is there any reason to use System.err here. Lets use log

Reply:
Ciphertool does not have a logger as a dependency

Reply:
Since it is an interactive tool, most of the lines should be written back to the command prompt for the user to interact with.